### DISCUSSION TITLE
server browser http://127.0.0.1:3000

### DISCUSSION THREAD

**Discussion by khaldun211** [stitionai/devika#334] [General]
Is this IP address correct for launching the server browser http://127.0.0.1:3000? iget refused to connect
The address localhost:3000 started the browser, but it did not react by creating a new project and once i insert a prompt i get the message to Create a new project.

**Comment by EveCrystali** [comment]
I was having the same issue. So this is what i've done (i've made progress but there is still work to do because i've reached out a new one 😄) :

**Diagnosing with Development Tools (F12):**

- **CORS Error**: 
  After looking with Development Tools (F12), I found out that the issue was due to a CORS Error:
  Encountered a CORS policy error while trying to connect the frontend at `http://localhost:3000` to a backend service at `http://127.0.0.1:1337`. The error was due to the absence of the `Access-Control-Allow-Origin` header.

**Proposed Solutions:**

1. **Adjust Backend Server Settings**:
   - Include CORS headers. For Node.js with Socket.IO, configure the server to accept requests from `http://localhost:3000`.

2. **Use Development Tools or Browser Extensions**:
   - Bypass CORS restrictions for testing purposes.

3. **Launch Browsers with Flags**:
   - Such as Chrome with `--disable-web-security`, temporarily disabling CORS for development testing.

**Security Considerations:**

- Emphasize the importance of using these methods only for local development and testing due to potential security risks.

**Testing and Validation:**

- Recommended restarting the backend server after configuring CORS and reloading the frontend application for testing.

**Current Status:**

- The solution worked for about 5 minutes, and then I got stuck. But at least it solves the issue of creating a project. However, now my model is not found anymore.

**New Issue - Console Log Error in Loop:**
```
socket__io-client.js?v=94e5ae89:898
GET http://127.0.0.1:1337/socket.io/?EIO=4&transport=polling&t=Owjujrt 426 (Upgrade Required)
```

**Reply by tblock-zz** [reply]
Hello, I am a newbee,
can you explain what I have to do?
I also see this 0.0.0.0:1337. But I do not understand from your post, what I have to do to fix this error.
I am using ubuntu 22.04.
When I start "python3 devika.py" I can see by netstat -aon | grep "1337" that 0.0.0.0:1337 is shown.
When I start "bun run dev", I see only "vite dev" with no http address.
Thanks
